olive/resource_path.py

```diff
@@ -107,32 +107,32 @@ def create_resource_path(self) -> ResourcePath:
         return ResourcePath.registry[self.type](self.config)
 
 
+VALID_RESOURCE_CONFIGS = (str, Path, dict, ResourcePathConfig, ResourcePath)
+OLIVE_RESOURCE_ANNOTATIONS = Optional[Union[str, Path, dict, ResourcePathConfig, ResourcePath]]
+
+
 def create_resource_path(
-    resource_path: Optional[Union[str, Path, Dict[str, Any], ResourcePathConfig, ResourcePath]]
+    resource_path: OLIVE_RESOURCE_ANNOTATIONS,
 ) -> Optional[Union[ResourcePath, List[ResourcePath]]]:
     """Create a resource path from a string or a dict.
 
-    If a string is provided, it is inferred to be a file, folder, or string name.
-    If a Path is provided, it is inferred to be a file or folder.
+    If a string or Path is provided, it is inferred to be a file, folder, or string name.
```

Comment:
This change might be more confusing.

Reply:
The logic did not change. the doc was not update when the behavior was changed by https://github.com/microsoft/Olive/pull/352 so I updated it to fix the doc. 

Looks like sometimes, this is being called with `Path` object even though it is a string name. Might be because of some validator somewhere but it is not related to this PR.